client typecheck fails on fresh checkout: stale @vicoop-bridge/protocol dist hides backendKind / optional agentCard

## Symptom

On a clean checkout of `main`, `pnpm --filter @vicoop-bridge/client typecheck` fails with:

```
src/client.ts(176,13): error TS2353: Object literal may only specify known properties, and 'backendKind' does not exist in type '{ type: "hello"; version: "0.1"; agentId: string; agentCard: { ... }; token: string; }'.
```

Note the type the compiler sees has `agentCard` as **required** and **no `backendKind`** at all — neither matches the current source.

## Root cause

`packages/protocol/package.json` resolves consumers through built artifacts:

```json
"main": "./dist/index.js",
"types": "./dist/index.d.ts",
"exports": {
  ".": {
    "types": "./dist/index.d.ts",
    "import": "./dist/index.js",
    "default": "./dist/index.js"
  }
}
```

`dist/` is gitignored, so on any fresh checkout (or whenever `pnpm install` runs without a follow-up build) consumers see whatever `dist/` last got compiled into the working tree — or nothing.

`packages/protocol/src/index.ts` was updated in `d3f9172` "Resolve agent cards server-side" to:

- make `HelloFrame.agentCard` optional (`AgentCard.optional()`)
- add `HelloFrame.backendKind: BackendKind.optional()`

The matching consumer change in `packages/client/src/client.ts:176` passes `backendKind` into the hello frame. But if the protocol's `dist/` was last built before `d3f9172`, the client's `tsc` reads a `.d.ts` where `agentCard` is required and `backendKind` doesn't exist, and rejects line 176.

A local `pnpm --filter @vicoop-bridge/protocol build` makes the typecheck pass cleanly, confirming the diagnosis.

## Why this is a real footgun, not just a one-off

- New contributors hit it on first checkout — `pnpm install && pnpm --filter @vicoop-bridge/client typecheck` should be green out of the box, and currently isn't.
- CI passes only because (presumably) something in the pipeline runs `pnpm -r build` before per-package typecheck. The repo-root `pnpm typecheck` (`pnpm -r typecheck`) does **not** depend on build.
- Anyone who edits `protocol/src/*` and runs `pnpm --filter @vicoop-bridge/client typecheck` without rebuilding protocol gets confusing errors that point at the **consumer** rather than the actual source-of-truth (the protocol source).

## Possible fixes (pick one — happy to implement whichever direction you prefer)

1. **Make protocol consumers resolve from source.** Switch `packages/protocol/package.json` `main` / `types` / `exports` to point at `src/index.ts`, and rely on tsx / project references for the runtime side. Eliminates the build-before-typecheck ordering hazard entirely. Likely the cleanest fix for a private workspace package.

2. **TypeScript project references.** Add `"references": [{ "path": "../protocol" }]` to `packages/client/tsconfig.json` (and server/admin-ui) and a matching `composite: true` on protocol. `tsc --build` then handles ordering automatically. More invasive but standard.

3. **Keep dist-based resolution, fix the workflow.** Add a `prebuild` / `pretypecheck` script on protocol consumers that runs `pnpm --filter @vicoop-bridge/protocol build` first, plus document it in CONTRIBUTING. Cheapest fix but the trap stays loaded.

## Repro

```bash
git checkout main
rm -rf packages/protocol/dist
pnpm install
pnpm --filter @vicoop-bridge/client typecheck
# → fails on src/client.ts:176

pnpm --filter @vicoop-bridge/protocol build
pnpm --filter @vicoop-bridge/client typecheck
# → passes
```

## Notes

- Discovered while running `pnpm --filter @vicoop-bridge/client typecheck` during work on #102 / #105. The error is unrelated to that change — it reproduces on stock `main`.
- `client.ts:176` itself is correct; no code change needed there once protocol resolution is fixed.

--- a/packages/protocol/package.json
+++ b/packages/protocol/package.json
@@ -4,16 +4,17 @@
   "private": true,
   "type": "module",
   "main": "./dist/index.js",
-  "types": "./dist/index.d.ts",
+  "types": "./src/index.ts",
   "exports": {
     ".": {
-      "types": "./dist/index.d.ts",
+      "types": "./src/index.ts",
       "import": "./dist/index.js",
       "default": "./dist/index.js"
     }
   },
   "files": [
-    "dist"
+    "dist",
+    "src"
   ],
   "scripts": {
     "build": "tsc -p tsconfig.json",

